engine/apps/zvonok/zvonok_phone_provider.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import requests
+import os
+
+from random import randint
+
+from django.core.cache import cache
+
+from apps.base.models import LiveSetting
+from apps.base.utils import live_settings
+
+from apps.phone_notifications.exceptions import (
+    FailedToFinishVerification,
+    FailedToMakeCall,
+    FailedToSendSMS,
+    FailedToStartVerification,
+)
+from apps.phone_notifications.phone_provider import PhoneProvider, ProviderFlags
+
+ZVONOK_CALL_URL = 'https://zvonok.com/manager/cabapi_external/api/v1/phones/call/'
+ZVONOK_TIMEOUT = os.environ.get("ZVONOK_TIMEOUT", default=30)
+ZVONOK_SPEAKER = os.environ.get("ZVONOK_SPEAKER", default="Joanna")
+
+logger = logging.getLogger(__name__)
+
+class ZvonokPhoneProvider(PhoneProvider):
+    """
+    ZvonokPhoneProvider is an implementation of phone provider which supports only voice calls (zvonok.com).
+    """
+
+    def make_notification_call(self, number, message):
+        logger.info(f"ZvonokProvider.make_notification_call to number: {number}")
+        try:
+            response = self._create_call(number, message)
+            if response.json().get("status") == "error":
+                logger.error(f"ZvonokPhoneProvider.make_notification_call FAILED to number: {number}")
+                raise FailedToMakeCall
+        except requests.exceptions.RequestException as e:
+            logger.error(f"ZvonokPhoneProvider.make_notification_call: failed {e}")
+            raise FailedToMakeCall
+
+    def make_call(self, number: str, message: str):
+        logger.info(f"ZvonokPhoneProvider.make_call to number: {number}")
+        try:
+            response = self._create_call(number, message)
+            if response.json().get("status") == "error":
+                logger.error(f"ZvonokPhoneProvider.make_call FAILED to number: {number}")
+                raise FailedToMakeCall
+        except requests.exceptions.RequestException as e:
+            logger.error(f"ZvonokPhoneProvider.make_call: failed {e}")
+            raise FailedToMakeCall
+
+    def make_verification_call(self, number: str):
+        code = str(randint(100000, 999999))
+        cache.set(self._cache_key(number), code, timeout=10 * 60)
+        codewspaces = " ".join(code)
+        try:
+            response = self._create_call(number, f"Your verification code is {codewspaces}")
+            if response.json().get("status") == "error":
+                logger.error(f"ZvonokPhoneProvider.make_verification_call FAILED to number: {number}")
+                raise FailedToStartVerification
+        except requests.exceptions.RequestException as e:
```

Comment:
Could you please catch requests.exceptions.ConnectionError here:
except (RequestException, ConnectionException) as e